### PR TITLE
Add number of iterations & add raise error when solver not found.

### PR DESCRIPTION
**Pull request by opieters**
Add variable to store number of iterations & add raise error when solver cannot be identified.

### CODE REVIEW

File: pyglmnet/pyglmnet.py [hunk 1]
```diff
@@ -768,6 +771,8 @@ def fit(self, X, y):
                     msg = ('\tConverged in {0:d} iterations'.format(t))
                     logger.info(msg)
                     break
+            else:
+                raise ValueError("Solver", self.solver, "not supported.")
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: Thanks for the contribution! Could you add a test?

File: pyglmnet/pyglmnet.py [hunk 2]
```diff
@@ -780,6 +785,9 @@ def fit(self, X, y):
             if callable(self.callback):
                 self.callback(beta)
 
+        if self.n_iter == self.max_iter:
+            logger.warning("Reached max number of iterations without conversion.")
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]:
```suggestion
            logger.warning("Reached max number of iterations without convergence.")
```

File: pyglmnet/pyglmnet.py [hunk 3]
```diff
@@ -747,6 +748,8 @@ def fit(self, X, y):
 
         # Iterative updates
         for t in range(0, self.max_iter):
+            self.n_iter += 1
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: can you add an underscore at the end to follow scikit-learn convention?

**Comment by jasmainak** [comment]
We follow pep8 conventions @opieters so Travis will not pass until you conform to that.

**Comment by opieters** [comment]
Ok, I will modify the code to follow the necessary conventions and try to add a test.

**Comment by opieters** [comment]
I think all of the issues are fixed now.

File: tests/test_pyglmnet.py [hunk 4]
```diff
@@ -436,3 +436,28 @@ def test_api_input_types_y():
...
+    with pytest.raises(ValueError):
+        glm.fit(X, y)
+
+    # test that normal solver is OK
+    glm = GLM(distr='gaussian', solver="batch-gradient")
+
+    # This would work without errors
+    glm.fit(X, y)
+    glm.predict(X)
+    glm.score(X, y)
```
Comment (jasmainak) [inline tests/test_pyglmnet.py]: can you add the test to an existing function instead of making a new one just for one test?

Comment (opieters) [reply tests/test_pyglmnet.py]: Done.

File: pyglmnet/pyglmnet.py [hunk 5]
```diff
@@ -780,6 +785,10 @@ def fit(self, X, y):
             if callable(self.callback):
                 self.callback(beta)
 
+        if self.n_iter_ == self.max_iter:
+            logger.warning(
+                "Reached max number of iterations without conversion.")
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: conversion -> convergence?

Can you also add a test that checks this warning is raised?

**Comment by jasmainak** [comment]
closed by #316

**Pull request closed without merging**